sdk/identity/identity/src/credentials/workloadIdentityCredential.ts

```diff
@@ -59,41 +166,191 @@ export class WorkloadIdentityCredential implements TokenCredential {
     );
+
     this.client = new ClientAssertionCredential(
       tenantId,
       clientId,
       this.readFileContents.bind(this),
-      options,
+      workloadIdentityCredentialOptions,
     );
   }
 
+  /**
+   * Creates a proxy HttpClient that intercepts token requests and redirects them to the Kubernetes endpoint
+   * Caching is handled at the credential level to persist across token requests
+   */
+  private createAksProxyClient(tokenEndpoint: string): HttpClient {
+    const defaultClient = createDefaultHttpClient();
+
+    return {
+      sendRequest: async (request: PipelineRequest): Promise<PipelineResponse> => {
+        const requestUrl = new URL(request.url);
+
+        logger.info(
+          `${credentialName}: Redirecting request to Kubernetes endpoint: ${tokenEndpoint}`,
+        );
+        const tlsSettings = await this.getTlsSettings();
+
+        // Rewrite request URL following Go rewriteProxyRequestURL pattern
+        const proxyUrl = new URL(tokenEndpoint);
+
+        // Remove leading slash from request path and join with proxy path
+        const requestPath = requestUrl.pathname.replace(/^\//, "");
+        const combinedPath = proxyUrl.pathname.endsWith("/")
+          ? proxyUrl.pathname + requestPath
+          : proxyUrl.pathname + "/" + requestPath;
+
+        // Create new URL preserving query and fragment from original request
+        const newUrl = new URL(proxyUrl.origin);
+        newUrl.pathname = combinedPath;
+        newUrl.search = requestUrl.search;
+        newUrl.hash = requestUrl.hash;
+
+        request.url = newUrl.toString();
+        request.tlsSettings = tlsSettings;
+
+        logger.info(`${credentialName}: Sending request to ${request.url}`);
+
+        // Forward the modified request with custom TLS settings
+        return defaultClient.sendRequest(request);
+      },
+    };
+  }
+
+  /**
+   * Gets TLS settings for the request.
+   * Handles a few scenarios with CA data or CA file provided.
+   */
+  private async getTlsSettings(): Promise<TlsSettings & { servername?: string }> {
+    // No CA overrides, use default transport
+    if (!this.caData && !this.caFile) {
+      if (!this.cachedTlsSettings) {
+        this.cachedTlsSettings = this.sniName ? { servername: this.sniName } : {};
+      }
+      return this.cachedTlsSettings;
+    }
+
+    // Host provided CA bytes in AZURE_KUBERNETES_CA_DATA and can't change now
+    if (!this.caFile) {
+      if (!validatePemCertificates(this.caData!)) {
+        throw new CredentialUnavailableError(
+          `${credentialName}: is unavailable. ${ErrorMessages.INVALID_CA_CERTIFICATES}`,
+        );
+      }
+      if (!this.cachedTlsSettings) {
+        this.cachedTlsSettings = this.sniName ? { servername: this.sniName } : {};
+        this.cachedTlsSettings.ca = this.caData;
+      }
```

Comment:
We can validate the cert bytes just once because they won't change
```suggestion
      if (!this.cachedTlsSettings) {
        if (!validatePemCertificates(this.caData!)) {
          throw new CredentialUnavailableError(
            `${credentialName}: is unavailable. ${ErrorMessages.INVALID_CA_CERTIFICATES}`,
          );
        }
        this.cachedTlsSettings = this.sniName ? { servername: this.sniName } : {};
        this.cachedTlsSettings.ca = this.caData;
      }
```